src/dev/build/tasks/bin/scripts/kibana

```diff
@@ -44,8 +40,27 @@ NODE="${DIR}/node/pointer-compression/bin/node"
 {{/linux}}
 test_node
 
+BASE_NODE_OPTIONS="--no-warnings --max-http-header-size=65536"
+if [ -f "${CONFIG_DIR}/node.options" ]; then
+  KBN_NODE_OPTS="$(grep -v ^# < ${CONFIG_DIR}/node.options | xargs)"
+fi
 # Enable capturing heap snapshots. See https://nodejs.org/api/cli.html#--heapsnapshot-signalsignal
 # For now only on POSIX platforms, term signals work differently on Windows.
 NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT="${NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT:---heapsnapshot-signal=SIGUSR2 --diagnostic-dir=$DIR/data}"
+ALL_NODE_OPTIONS="$BASE_NODE_OPTIONS $NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT $KBN_NODE_OPTS $NODE_OPTIONS"
+
+if [ "$ELASTIC_CONTAINER" = "true" ] && ! echo "$ALL_NODE_OPTIONS" | grep -q -- '--max-old-space-size'; then
+  max_old_space_percentage=75
+  heap_mb=
+  cgroup_memory_limit_bytes=$(cat /sys/fs/cgroup/memory.max 2>/dev/null || cat /sys/fs/cgroup/memory/memory.limit_in_bytes 2>/dev/null)
+  mem_available_kb=$(grep -E '^MemAvailable:' /proc/meminfo 2>/dev/null | grep -Eo '[0-9]+')
```

Comment:
Some POSIX oddities: prefer `printf '%s' "$ALL_NODE_OPTIONS" | grep ...` over `echo` and consider `awk '/^MemAvailable:/ {print $2}' /proc/meminfo` instead of `grep -Eo` for portability.